### PR TITLE
Issue 21716 anon upload

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/ContentResource.java [hunk 1]
```diff
@@ -1671,35 +1680,42 @@ private void processFile(final Contentlet contentlet,
...
-                    fieldVarName = binaryFields.remove(0);
+        try(final InputStream input = part.getEntityAs(InputStream.class)){
+            final String tmpFileName = part.getContentDisposition().getFileName();
+            final String filename = FileUtil.sanitizeFileName(tmpFileName);
+            if(!tmpFileName.equals(filename)) {
+                SecurityLogger.logInfo(getClass(), "Invalid filename uploaded, possible exploit attempt: " + tmpFileName);
+                throw new DotSecurityException("Invalid filename uploaded : " + tmpFileName);
+            }
+            
+            final File tmpFolder = new File(APILocator.getFileAssetAPI().getRealAssetPathTmpBinary() + UUIDUtil.uuid());
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentResource.java:1691]: We should encapsulate the logic into:
```
 the File file = FileAssetAPI.copyToFile (input);
 ```

File: dotCMS/src/main/java/com/dotcms/rest/api/MultiPartUtils.java [hunk 2]
```diff
@@ -110,7 +119,12 @@ public MultiPartUtils(final FileAssetAPI fileAssetAPI) {
                 throw new IOException("Unable to create temp folder to save binaries");
             }
 
-            final String filename = part.getContentDisposition().getFileName();
+            final String tmpFileName = part.getContentDisposition().getFileName();
+            final String filename = FileUtil.sanitizeFileName(tmpFileName);
+            if(!tmpFileName.equals(filename)) {
+                SecurityLogger.logInfo(getClass(), "Invalid filename uploaded, possible exploit attempt: " + tmpFileName);
+                throw new DotRuntimeException("Invalid filename uploaded : " + tmpFileName);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/MultiPartUtils.java:126]: This should be Illegal Argument Exception

File: dotCMS/src/main/java/com/dotmarketing/util/FileUtil.java [hunk 3]
```diff
@@ -161,27 +163,54 @@ public static String getAbsoluteFileAssetPath(String inode, String extenstion) {
...
-    StringBuilder cleanName = new StringBuilder();
-    int len = badFileName.codePointCount(0, badFileName.length());
-    for (int i = 0; i < len; i++) {
-      final int c = badFileName.codePointAt(i);
-      if (Arrays.binarySearch(illegalChars, c) < 0) {
-        cleanName.appendCodePoint(c);
+  public static String sanitizeFileName(final String badFileNameIncoming) {
+      
+      
+      final String fileExtention = UtilMethods.isSet(UtilMethods.getFileExtension(badFileNameIncoming))
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/FileUtil.java:183]: nice

**Review by jdotcms — APPROVED** [review]

**Comment by wezell** [comment]
Closed in favor of Issue: #21791  PR: #21792

**Pull request closed without merging**